File: basic/armstrong.go
package basic

import (
	"fmt"
	"math"
)

func armstrong() {

	input := 153
	lengthCheck := input
	original := input
	length := 0
	result := 0

	for lengthCheck > 0 {
		lengthCheck = lengthCheck / 10
		length++
	}

	for input > 0 {
		lastDigit := input % 10
		result += int(math.Pow(float64(lastDigit), float64(length)))
		input = input / 10
	}

	if original == result {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}

}

File: c#/basic/armstrong.cs
using System;

public class HelloWorld
{
	public static void Main(string[] args)
	{
		int input1 = 153;
		
		if (isArmStrong(input1))
		{
			Console.WriteLine("Yes Armstrong");
		}
		else
		{
			Console.WriteLine("Not Armstrong");
		}
	}

	static bool isArmStrong(int input1)
	{
		int result = 0;
		var length = input1.ToString().Length;
		var originalNumber = input1;

		while(input1 > 0)
		{
			var lastDigit = input1 % 10;
			result = result + (int)Math.Pow(lastDigit, length);
			input1 = input1/10;
		}

		return result == originalNumber;
	}
}

